src/server/pfs/server/env.go

```diff
@@ -41,17 +43,33 @@ type Env struct {
 }
 
 func EnvFromServiceEnv(env serviceenv.ServiceEnv, txnEnv *txnenv.TransactionEnv) (*Env, error) {
-	// Setup etcd, object storage, and database clients.
-	objClient, err := obj.NewClient(env.Context(), env.Config().StorageBackend, env.Config().StorageRoot)
-	if err != nil {
-		return nil, err
+	cfg := env.Config()
+
+	// Only setup the new GoCDK bucket if necessary.  The storage layer will prefer it, if it is non-nil.
+	// Disabling GoCDK will completely bypass the entire setup, and revert to the old obj.Client.
+	// This is designed to be a quick fix in the field, if issues are discovered with Go CDK.
+	var objClient obj.Client
+	var bucket *obj.Bucket
+	if cfg.GoCDKEnabled {
+		var err error
+		bucket, err = obj.NewBucket(env.Context(), cfg.StorageBackend, cfg.StorageRoot)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		var err error
+		objClient, err = obj.NewClient(env.Context(), cfg.StorageBackend, cfg.StorageRoot)
+		if err != nil {
+			return nil, err
+		}
 	}
 	etcdPrefix := path.Join(env.Config().EtcdPrefix, env.Config().PFSEtcdPrefix)
 	if env.AuthServer() == nil {
 		panic("auth server cannot be nil")
 	}
 	return &Env{
 		ObjectClient: objClient,
+		Bucket:       bucket,
```

Comment:
We always pass bucket through, and the storage layer will try to use it, if it is available.